docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+The sections below give more details of each component.
+
+#### UserInterface component
+
+![userinterfacecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/UserInterfaceComponent.png)
+
+Figure 2. Class Diagram of the UserInterface component
+
+API: UserInterface.java
+
+The UserInterface component,
+* Prompts commands from the user.
+* Execute user commands using the Controller component.
+* Displays information based on changes to Model data.
+* Prints the user’s events of the week in a timetable format.
+
+This component uses the singleton design, meaning that there is only an instance required throughout the entire lifetime of the application, obtained with the following command
+`UserInterface userInterface = UserInterface.getInstance()`
+<br />
+<br />
+
+#### Controller component
+
+![controllercomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ControllerComponent.png)
+
+Figure 3. Class Diagram of the Controller component
+
+API: ControlManager.java
+
+The Controller component,
+* Receives user input from the user interface.
+* Extracts the command and model type from the user input.
+* Generate the required command from the CommandFactory.java class
+* Extract the required model to be accessed and modified.
+* Executes the command with the corresponding model.
+
+<br />
+<br />
+
+#### Model component
+
+![modelcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ModelComponent.png)
+
+Figure 4. Class Diagram of the Model component
+
+API: Model.java
+
+The Model component,
+* Holds all the in-memory data of type event, quiz and contact.
+* Each data type has a corresponding manager that the controller can interface with. This is named as <datatype>Manager.java.
+* During a command execution, the manager will handle the modification and reading of its data type.
+* After the execution, the corresponding output is then passed on to the User Interface component to be shown to the user.
+
+<br />
+<br />
+
+#### Storage component
+![storagecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/StorageComponent.png)
+
+Figure 5. Class Diagram of the Storage component
+
+API: StorageManager.java
+
+The Storage component,
+* Creates the necessary data files for the operation of Plan&Score.
+* Reads encoded data from Plan&Score’s data files.
+* Writes encoded data to Plan&Score’s data files.
+
+There are 3 categories of data stored by Plan&Score: `event`, `quiz` and `config`
+
+Plan&Score loads data automatically from .txt files in the `data` directory.
+Each `StorageManager` reads in their respective data files through a `decoder` and writes to the same file through an `encoder`
+
+##### Event Storage
+![eventstorage](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/EventStorageManager.png)
```

Comment:
![image](https://user-images.githubusercontent.com/47527482/97396594-5aed0f80-1922-11eb-9821-c52b03208e4a.png)
Maybe your arrow for inheritance is pointed in the wrong direction
maybe you can add an association label for creates